app/src/main/java/com/google/android/soundchecker/dsd/DsfReader.kt

```diff
@@ -147,9 +148,15 @@ class DsfReader(inputStream: InputStream) {
     private fun readNextBlock(channel: Int): Boolean {
         Arrays.fill(mData[channel], 0, mData[channel].size - 1, 0x0.toByte())
         try {
+            if (mDataLeft < mData[channel].size && mStream.markSupported()) {
+                Log.d(TAG, "Reach end of file, mark supported, reset and play from beginning")
+                mStream.reset()
+                mDataLeft = mDataTotal
+            }
             val sz = mStream.read(mData[channel])
             if (sz != mData[channel].size) {
                 Log.w(TAG, "Cannot read full buffer: $sz")
+                return false
```

Comment:
Do we want to support reading a partial buffer at the end?
This seems expected?
Maybe return false if nothing read. Or return size read.